src/main/kotlin/io/bazel/kotlin/plugin/jdeps/BaseJdepsGenExtension.kt

```diff
@@ -16,23 +16,69 @@ abstract class BaseJdepsGenExtension(
   ) {
     val directDeps = configuration.getList(JdepsGenConfigurationKeys.DIRECT_DEPENDENCIES)
     val targetLabel = configuration.getNotNull(JdepsGenConfigurationKeys.TARGET_LABEL)
-    val explicitDeps = createDepsMap(explicitClassesCanonicalPaths)
+    val fullClasspath = configuration.getList(JdepsGenConfigurationKeys.FULL_CLASSPATH)
 
-    doWriteJdeps(directDeps, targetLabel, explicitDeps, implicitClassesCanonicalPaths)
+    // Create mapping from canonical paths to original classpath paths
+    val canonicalToClasspath = createCanonicalToClasspathMap(fullClasspath)
+
+    val explicitDeps = createDepsMap(explicitClassesCanonicalPaths, canonicalToClasspath)
+
+    doWriteJdeps(
+      directDeps,
+      targetLabel,
+      explicitDeps,
+      implicitClassesCanonicalPaths,
+      canonicalToClasspath,
+    )
 
     doStrictDeps(configuration, targetLabel, directDeps, explicitDeps)
   }
 
+  /**
+   * Creates a mapping from canonical filesystem paths to original classpath paths.
+   * This ensures jdeps uses the same paths that were on the compiler's classpath.
+   * When multiple classpath entries point to the same canonical file, prefer runfiles paths.
+   */
+  private fun createCanonicalToClasspathMap(classpathEntries: List<String>): Map<String, String> {
+    val mapping = mutableMapOf<String, String>()
```

Comment:
Are these jdeps changes related to this supporting Kotlin 2.2 or is this fixing something independent?